main.go

```diff
@@ -282,3 +289,17 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func mapCRDAvailability(ctx context.Context, cl apiclient.Client) (map[string]bool, error) {
+	crdExist := map[string]bool{}
+	crdList := &metav1.PartialObjectMetadataList{}
+	crdList.SetGroupVersionKind(extv1.SchemeGroupVersion.WithKind("CustomResourceDefinitionList"))
+	if err := cl.List(ctx, crdList); apiclient.IgnoreNotFound(err) != nil {
+		return nil, fmt.Errorf("error getting CRD, %v", err)
+	}
+	// Iterate over the list and populate the map
+	for _, crd := range crdList.Items {
+		crdExist[crd.Name] = true
+	}
```

Comment:
CRDs are very large objects, this loop will creat a copy of each CRD in the items list, please change to.
 

```suggestion
	for i := range crdList.Items {
		crdExist[crdList.Items[i].Name] = true
	}
```